frostsnap_core/src/device.rs

```diff
@@ -69,36 +63,49 @@ impl FrostSigner {
         DeviceId::new(self.keypair().public_key())
     }
 
-    pub fn generate_nonces(
-        &mut self,
-        aux_rand: [u8; 32],
-        start: usize,
-        n: usize,
+    fn generate_nonces(
+        // this is always the device key for now but because of lifetimes issues it has to be passed in
+        secret_key: &Scalar,
+        start: u64,
     ) -> impl Iterator<Item = NonceKeyPair> + '_ {
         let mut nonce_rng = derive_nonce_rng! {
             nonce_gen => nonce::Deterministic::<Sha256>::default().tag(b"frostsnap/nonces"),
-            secret => self.keypair.secret_key(),
-            public => [aux_rand],
+            secret => secret_key,
+            public => [b""],
```

Comment:
Why did we drop `aux_rand` here? It certainly was annoying, but are we now deriving the same nonces for every key? I guess this is fine with a global device nonce_counter?

Reply:
`aux_rand` used to be there so we got a different set of nonces per key we generated. Now we don't nonces per key so it no longer makes sense. We will have different nocnes per coordinator but it was premature to introduce this before supporting a concept of coordinator identity.